src/backend/common/UsageTracker.h

```diff
@@ -0,0 +1,81 @@
+// Copyright 2017 The NXT Authors
```

Comment:
That's a good point, it is a bit simpler. Right now it has its own implementation in the state tracking which isn't that long (though it keeps getting longer). Merging it into the resource objects was another approach I thought of. That's fine as long as we're careful not to leak transient state between command buffer validations.